bfillings/mafft_v7.py

```diff
@@ -0,0 +1,356 @@
+            lines.append(''.join(['>', str(i+1)]))
+            lines.append(s)
+        return self._input_as_lines(lines)
+
+    def _tree_out_filename(self):
+        if self.Parameters['--treeout'].isOn():
+            tree_filename = self._absolute(str(self._input_filename))+'.tree'
+        else:
+            raise ValueError("No tree output file specified.")
+        return tree_filename
+
+    def getHelp(self):
+        """Method that points to the Mafft documentation."""
+
+        help_str = ("See Mafft documentation at:\n"
+                    "http://align.bmr.kyushu-u.ac.jp/"
+                    "mafft/software/manual/manual.html")
+        return help_str
+
+    def _get_result_paths(self, data):
+        result = {}
+        if self.Parameters['--treeout'].isOn():
+            out_name = self._tree_out_filename()
+            result['Tree'] = ResultPath(Path=out_name, IsWritten=True)
+        return result
+
+
+def align_unaligned_seqs(seqs_fp, moltype=DNA, params=None, accurate=False):
+    """Aligns unaligned sequences
+
+    Parameters
+    ----------
+    seqs_fp : string
+        file path of the input fasta file
+    moltype : callable constructor
+        It can be ``DNA``, ``RNA``, ``Protein`` classes that are defined in
+        ``skbio``. Default is DNA
+    params : dict-like type
+        It pass the additional parameter settings to the application.
+        Default is None.
+    accurate : boolean
+        Perform accurate alignment or not. It will sarificing performance
+        if set to True. Default is False.
+
+    Returns
+    -------
+    Alignment object
+        The aligned sequences.
+
+    See Also
+    --------
+    skbio.Alignment
+    skbio.DNA
+    skbio.RNA
+    skbio.Protein
+    """
+    # Create Mafft app.
+    app = Mafft(InputHandler='_input_as_path', params=params)
+
+    # Turn on correct sequence type
+    app.Parameters[MOLTYPE_MAP[moltype]].on()
+
+    # Do not report progress
+    app.Parameters['--quiet'].on()
+
+    # More accurate alignment, sacrificing performance.
+    if accurate:
+        app.Parameters['--globalpair'].on()
+        app.Parameters['--maxiterate'].Value = 1000
+
+    # Get results using int_map as input to app
+    res = app(seqs_fp)
+
+    # Get alignment as dict out of results
+    alignment = Alignment.read(res['StdOut'], constructor=moltype)
+
+    # Clean up
+    res.cleanUp()
+
+    return alignment
+
+
+def add_seqs_to_alignment(seqs_fp, aln_fp, moltype=DNA,
+                          params=None, accurate=False):
+    """Returns an Alignment object from seqs and existing Alignment.
+
+    The "--seed" option can be used for adding unaligned sequences into
+    a highly reliable alignment (seed) consisting of a small number of
+    sequences.
+
+    Parameters
+    ----------
+    seqs_fp : file path of the unaligned sequences
+
+    aln_fp : file path of the seed alignment
+
+    params : dict of parameters to pass in to the Mafft app controller.
+
```

Comment:
remove space between parameter declarations, add definition for parameters moltype and accurate